daemon/cluster/executor/container/controller.go

```diff
@@ -153,10 +154,40 @@ func (r *controller) Wait(pctx context.Context) error {
 	ctx, cancel := context.WithCancel(pctx)
 	defer cancel()
 
-	err := r.adapter.wait(ctx)
-	if ctx.Err() != nil {
+	// error chan for wait
+	waitErrCh := make(chan error)
+	// error chan for health check
+	healthErrCh := make(chan error)
+
+	runner := func(f func(ctx context.Context) error, errCh chan<- error) {
+		err := f(ctx)
+		select {
+		case <-r.closed:
+		case <-ctx.Done():
+		case errCh <- err:
+		}
+	}
+
+	// run wait and checkHealth
+	go runner(r.adapter.wait, waitErrCh)
+	go runner(r.checkHealth, healthErrCh)
```

Comment:
I will update it. The previous concern is how to sync with `shutdown failed on unhealthy` error correctly. 
Now it's much easier since we just log `shutdown failed on unhealthy` error.